flake.nix

```diff
@@ -5,27 +5,67 @@
     nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
   };
 
-  outputs = { self, flake-utils, naersk, nixpkgs }:
+  outputs = { self, flake-utils, naersk, nixpkgs, }:
     flake-utils.lib.eachDefaultSystem (system:
       let
-        pkgs = (import nixpkgs) {
-          inherit system;
-        };
-        naersk' = pkgs.callPackage naersk {};
+        pkgs = (import nixpkgs) { inherit system; };
+        inherit (pkgs) lib;
+
+        naersk' = pkgs.callPackage naersk { };
         nativeBuildInputs = with pkgs; [ pkg-config ];
+
+        darwinFrameworks = with pkgs.darwin.apple_sdk.frameworks; [
+          Security
+          SystemConfiguration
+        ];
+
+        devTools = with pkgs; [ cargo rustc ];
+
+        dependencies = with pkgs;
+          [ libgit2 openssl ]
+          ++ lib.lists.optionals stdenv.isDarwin darwinFrameworks;
+
+        envVars = {
+          LIBGIT2_NO_VENDOR = 1;
+          OPENSSL_NO_VENDOR = 1;
+        };
       in rec {
-        packages.default = naersk'.buildPackage {
+        # NB: if this does not build and you need to modify the file,
+        #     please ensure you also make the corresponding changes in the devshell
+        packages.default = naersk'.buildPackage ({
           inherit nativeBuildInputs;
           src = ./.;
-          buildInputs = with pkgs; [ openssl ] ++ (pkgs.lib.lists.optionals (stdenv.isDarwin) [ darwin.apple_sdk.frameworks.SystemConfiguration ]);
-        };
-        devShells.default = pkgs.mkShell {
-          nativeBuildInputs = with pkgs; nativeBuildInputs ++ [
-            cargo
-            rustc
-          ];
+          buildInputs = devTools ++ dependencies;
+        } // envVars);
+
+        devShells.default = pkgs.mkShell ({
+          buildInputs = nativeBuildInputs ++ devTools ++ dependencies;
+        } // envVars);
+
+        formatter = with pkgs;
+          writeShellApplication {
+            name = "nixfmt-nix-files";
+            runtimeInputs = [ fd nixfmt ];
+            text = "fd \\.nix\\$ | xargs nixfmt";
+          };
```

Comment:
Drawing attention to the new formatter script in case this is different to what will end up in helpkgs